locker/locking.go

```diff
@@ -0,0 +1,73 @@
+package locker
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+)
+
+const (
+	UpdateSleepTime = 2 * time.Second
+	ValidTimePeriod = 5 * time.Second
+)
+
+type LockInformation struct {
+	ID   string
+	Key  string
+	Date time.Time
+}
+
+func (l *LockInformation) Unmarshal(data []byte) error {
+	var tmp struct {
+		ID   string
+		Key  string
+		Date string
+	}
+	if err := json.Unmarshal(data, &tmp); err != nil {
+		return err
+	}
+
+	if tmp.ID == "" {
+		return fmt.Errorf("lock information is missing ID")
+	}
+	if tmp.Key == "" {
+		return fmt.Errorf("lock information is missing Key")
+	}
+
+	l.ID = tmp.ID
+	l.Key = tmp.Key
+
+	// Parse the date string with UTC time zone
+	parsedTime, err := time.Parse(time.RFC3339, tmp.Date)
+	if err != nil {
+		return err
+	}
+	l.Date = parsedTime.UTC()
+
+	return nil
+}
+
+func (l *LockInformation) Marshal() ([]byte, error) {
+	return json.Marshal(l)
+}
+
+type MultiLock interface {
+	Lock(lock string) (Key, error)
+	Unlock(key Key) error
+}
+
+type Key interface {
+	ID() string
+	Key() string
+	ExpirationTime() time.Time
+	SetExpirationTime(time.Time) error // CHECK: should this be here?
+}
+
+// A method used to close a channel only once
+func closeOnce(ch chan error) {
```

Comment:
nit: instead of the error type, maybe an empty struct{}. I feel like seeing "error" is throwing me off.

Reply:
removing it since it was from previous interface design